src/wrapper/objects/auto_long_array.rs

```diff
@@ -0,0 +1,85 @@
+use std::ptr::NonNull;
+
+use log::debug;
+
+use crate::{errors::*, JNIEnv, objects::JObject};
+use crate::objects::release_mode::ReleaseMode;
+use crate::sys::jlong;
+
+/// Auto-release wrapper for pointer-based long arrays.
+///
+/// This wrapper is used to wrap pointers returned by get_long_array_elements.
+///
+/// These arrays need to be released through a call to release_long_array_elements.
+/// This wrapper provides automatic array release when it goes out of scope.
+pub struct AutoLongArray<'a: 'b, 'b> {
+    obj: JObject<'a>,
+    ptr: NonNull<jlong>,
+    mode: ReleaseMode,
+    is_copy: bool,
+    env: &'b JNIEnv<'a>,
+}
+
+impl<'a, 'b> AutoLongArray<'a, 'b> {
+    /// Creates a new auto-release wrapper for a pointer-based long array
+    ///
+    /// Once this wrapper goes out of scope, `release_long_array_elements` will be
+    /// called on the object. While wrapped, the object can be accessed via
+    /// the `From` impl.
```

Comment:
Given clients don't use this method directly, would it make sense to merge this doc into the struct level docs, that are visible?